.changeset/breezy-rats-kick.md

```diff
@@ -0,0 +1,123 @@
+'@reown/appkit-scaffold-ui': patch
+'@reown/appkit-adapter-bitcoin': patch
+'@reown/appkit-adapter-ethers': patch
+'@reown/appkit-adapter-ethers5': patch
+'@reown/appkit-adapter-solana': patch
+'@reown/appkit-adapter-wagmi': patch
+'@reown/appkit': patch
+'@reown/appkit-utils': patch
+'@reown/appkit-cdn': patch
+'@reown/appkit-cli': patch
+'@reown/appkit-codemod': patch
+'@reown/appkit-common': patch
+'@reown/appkit-core': patch
+'@reown/appkit-experimental': patch
+'@reown/appkit-pay': patch
+'@reown/appkit-polyfills': patch
+'@reown/appkit-siwe': patch
+'@reown/appkit-siwx': patch
+'@reown/appkit-testing': patch
+'@reown/appkit-ui': patch
+'@reown/appkit-wallet': patch
+---
+
+Upgraded wallet button to support multichain via the `namespace` prop
+
+**Example usage with Components**
+
+```tsx
+import '@reown/appkit-wallet-button'
+
+const wallets = [
+  { wallet: 'metamask', namespace: 'eip155', label: 'MetaMask EVM' },
+  { wallet: 'metamask', namespace: 'solana', label: 'MetaMask Solana' },
+  { wallet: 'phantom', namespace: 'bip122', label: 'Phantom Bitcoin' },
+]
+
+export function WalletButtons() {
+  return (
+    <>
+      {wallets.map(({ wallet, namespace, label }) => (
+        <appkit-wallet-button
+          key={`${wallet}-${namespace}`}
+          wallet={wallet}
+          namespace={namespace}
+        />
+      ))}
+    </>
+  )
+}
+```
+
+**Example usage with Hooks**
+```tsx
+import { useAppKitWallet } from '@reown/appkit-wallet-button/react'
+
+export function YourApp() {
+  const { data, error, isPending, isSuccess, isError, connect } = useAppKitWallet({
+    namespace: 'eip155', // Use 'solana' or 'bip122' for other chains
+    onError: err => {
+      // ...
+    },
+    onSuccess: data => {
+      // ...
+    }
+  })
+
+  return (
+    <>
+      <button onClick={() => connect('walletConnect')}>Open QR modal</button>
+      <button onClick={() => connect('metamask')}>Connect to MetaMask</button>
+      <button onClick={() => connect('google')}>Connect to Google</button>
+    </>
+  )
+}
+```
+
+**Example usage with Vanilla JS**
+```html
+<!-- index.html -->
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <script type="module">
+      import '@reown/appkit-wallet-button'
+      import { createAppKitWalletButton } from '@reown/appkit-wallet-button'
+
+      const wallet = createAppKitWalletButton({ namespace: 'eip155' })
+
+      async function setup() {
+        if (!(await wallet.isReady())) {
+          wallet.subscribeIsReady(state => {
+            if (state.isReady) enableButtons()
+          })
+        } else {
+          enableButtons()
+        }
+      }
+
+      function enableButtons() {
+        document.querySelectorAll('button[data-wallet]').forEach(button => {
+          button.disabled = false
+          button.addEventListener('click', () => {
+            const walletId = button.getAttribute('data-wallet')
+            wallet.connect(walletId)
+          })
+        })
+      }
+
+      setup()
```

Comment:
Can we somehow make this setup much easier for users? Maybe moving this setup, and `enableButton()` logics inside and just return `enable` and `loading` etc props to the user?